crates/sui-graphql-rpc/schema/current_progress_schema.graphql

```diff
@@ -3859,11 +3859,19 @@ type TransactionBlock {
 type TransactionBlockConnection {
 	"""
 	Information to aid in pagination.
+	
+	When `scanLimit` is specified on `TransactionBlock` queries:
+	- Even if the current page is empty, `hasNextPage` and `hasPreviousPage` will be true until
+	the caller paginates to the end of either range.
+	
+	When `scanLimit` is not specified, then:
+	- `hasNextPage` and `hasPreviousPage` behave conventionally, returning false
+	when there are no more results in the respective direction.
+	
+	This approach ensures consistent pagination behavior, allowing traversal through potentially
+	sparse data sets within the specified or default range.
```

Comment:
Took a stab at rewording this for clarity (this documentation may be better placed on the query functions themselves, it's more duplication, but the connection type doesn't include a reference to `scanLimit`, so it's a little confusing to see it mentioned here).

> The `scanLimit` restricts the number of candidate transactions scanned for potential results. This can result in pages that have fewer than the requested `first` results when paginating forward (`last` when paginating backwards), but that still indicate that they have a next (or previous) page, indicating the scan limit was reached before the page size.
>
> In this case the `endCursor` (or `startCursor`) will point to the last transaction that was scanned (even if it is not a result). Requesting the next (or previous) page after this cursor will resume the search, scanning the next `scanLimit` many transactions in the direction of pagination. This process will continue until the full transaction range is exhausted.
>
> If no other filters are provided the transaction range consists of all transactions that occurred on chain, since genesis. It can be restricted using the `before` and `after` pagination cursors, as well as checkpoint bounds (`beforeCheckpoint`, `afterCheckpoint`, and `atCheckpoint`).